### PR TITLE
Make cancelling more reliable

### PR DESCRIPTION
**Pull request by coolo**
This should fix 95% of https://progress.opensuse.org/issues/12178

### CODE REVIEW

File: isotovideo [hunk 1]
```diff
@@ -410,16 +409,17 @@ if ($testfd) {
     kill_autotest;
 }
 
-diag "isotovideo done" unless $r;
-diag "FAIL" if $r;
+diag "isotovideo " . ($r ? 'failed' : 'done');
```
Comment (okurz) [inline isotovideo]: 'failed' : 'succeeded' ?

File: isotovideo [hunk 2]
```diff
@@ -410,16 +409,17 @@ if ($testfd) {
...
-    $clean_shutdown = $bmwqemu::backend->_send_json({cmd => 'is_shutdown'});
-    diag "BACKEND SHUTDOWN $clean_shutdown";
-};
-
-bmwqemu::stop_vm();
+if (!$r) {
+    eval {
+        $clean_shutdown = $bmwqemu::backend->_send_json({cmd => 'is_shutdown'});
+        diag "BACKEND SHUTDOWN $clean_shutdown";
+    };
```
Comment (okurz) [inline isotovideo]: The exception is never handled but should be, at least in a log output

File: isotovideo (hunk 1 above)
Comment (coolo) [inline isotovideo]: we don't know the job state at this point (and it doesn't matter) - succeeded expresses more than we know

File: isotovideo (hunk 2 above)
Comment (coolo) [inline isotovideo]: I disagree

**Comment by okurz** [comment]
LGTM

**Pull request merged**